auth-server/src/services/auth.ts

```diff
@@ -83,6 +85,40 @@ export default class AuthService {
 		return this.getSignedToken(user)
 	}
 
+	public async ChangePassword(token: string, oldPassword: string, newPassword: string) {
+		// verify a token symmetric - synchronous
+		const decoded = jwt.verify(token, process.env.ENCRYPTION_KEY)
```

Comment:
I wonder how much it makes sense to check the JWT here rather that sending the user id as post variable. Checking the refresh token + the old password will be stronger or equal to checking the JWT because a stoler old password would let an attacker login and get the refresh token anyways. Also we never use the jwt it in the auth server, so the front-end would have to deal with that unnecessarily.

Reply:
done